braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java

```diff
@@ -397,12 +435,84 @@ public Braze getUnderlyingInstance() {
         return braze;
     }
 
+    //////////////////////////////////////////////////
+    // UTIL Methods
+    //////////////////////////////////////////////////
     private static Date dateFromString(String date) {
         SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
         try {
             return formatter.parse(date);
         } catch (Exception e) {
+            RudderLogger.logError(e);
             return null;
         }
     }
+
+    // compare two address objects and return accordingly
+    private static boolean compareAddress(RudderTraits.Address curr, RudderTraits.Address prev) {
+        if (prev == null && curr != null) {
+            return true;
+        }
+
+        return prev != null && prev.getCity().equals(curr.getCity())
```

Comment:
Should be compared with current.
` return prev != null && prev.getCity().equals(curr.getCity())
                && prev.getCountry().equals(curr.getCountry())
                && prev.getPostalCode().equals(curr.getPostalCode())
                && prev.getState().equals(curr.getState())
                && prev.getStreet().equals(curr.getStreet());`